blog/content/post/2023-09-27-aem_phase2.md

```diff
@@ -0,0 +1,374 @@
+AEM with MFA to the internal disk where Qubes OS is installed. This is
+generally not recommended, because using MFA and internal disk does not
+provide any extra security over not using MFA. But it is just for
+demonstration purposes. So in your case feel free to install AEM even on an
+external storage. In our case the boot partition is `/dev/sda1` (yours may be
+different depending on where would you like to install AEM).
+
+```bash
+sudo anti-evil-maid-install -m /dev/sda1
+```
+
+![AEM installation 1](/img/qubes_aem_v0.2_11.png)
+
+![AEM installation 2](/img/qubes_aem_v0.2_12.png)
+
+If you chose MFA, then you will be presented a QR code to scan with an
+authentication app of your choice. it will show you a TOTP code later at boot
+process. This step will also ask you to provide a password for the secret and
+second time to provide any of the disk passwords for confirmation.
+
+The system now is ready to perform AEM boot, so reboot your machine!
+Now you should be presented with a SRK password prompt like this:
+
+![AEM SRK password prompt](/img/qubes_aem_v0.2_13.png)
+
+And later with the standard disk password prompt:
+
+![AEM disk password prompt](/img/qubes_aem_v0.2_14.png)
+
+After that AEM will print the messages saying that the secrets have been
+sealed to the LUKS PCR 13 DRTM PCRs 17 and 18:
+
+![AEM secrets sealing](/img/qubes_aem_v0.2_15.png)
+
+You should see the above screen only once (unless you change something with
+GRUB, Xen or dom0 kernel/initrd). Now that the AEM knows what the final PCR
+values should be, after next reboot AEM will successfully unseal the secrets
+it just sealed before a moment, and present a TOTP code. But before that, save
+the current PCR values by invoking the following command in dom0 terminal:
+
+```bash
+sudo tpm2_pcrread sha256
+```
+
+![TPM PCRs](/img/qubes_aem_v0.2_16.png)
+
+Note the PCR values with `sudo tpm2_pcrread sha256 > drtm_pcrs.txt` for
+example and reboot the machine. If everything went correctly, the PCRs should
+not change after reboot (you can comapre by reading the PCRs again with
+`tpm2_pcrread` to a file and `diff` command when booted) and you should be
+presented a TOTP code during LUKS key file password prompt, like this:
+
+![AEM TOTP code](/img/qubes_aem_v0.2_17.png)
+
+Check the TOTP code in your authenticator app to ensure it is correct and then
+type the password provided in the `anti-evil-maid-install` step. If there was
+any error, it should be displayed in the top left corner of the screen and you
+would be prompted for your regular disk password, because the LUKS key would
+not be unsealed by AEM.
+
+### Troubleshooting
+
+Sometimes the TOTP code does not immediately appear after second reboot,
+because the AEM secret could not be unsealed. It may be caused by the TPM
+safety timer. If your machine was not safely shut down (e.g. due to a power loss) and it had been running for less than approximately 70 minutes
```

Comment:
Wrap

Reply:
https://github.com/3mdeb/news-and-ideas/pull/543/commits/23be3297f4ca42f04cfc9231a9242525abf008d2#diff-c2e0e25a477d5cc5bf3075df551332eee15aaba457379ef35e8547639f285757R342